### PR TITLE
Add HTTP to gRPC auth. credential forwarding

### PR DESCRIPTION
**Pull request by t-chaik**
See discussion at https://github.com/buildbarn/bb-browser/issues/5.

### CODE REVIEW

File: cmd/bb_browser/browser_service.go [hunk 1]
```diff
@@ -49,6 +50,15 @@ func getDigestFromRequest(req *http.Request) (*util.Digest, error) {
 		})
 }
 
+func extractContextFromRequest(req *http.Request) context.Context {
+	ctx := req.Context()
+	md := metautils.ExtractIncoming(ctx)
+	if req.Header.Get("Authorization") != "" {
```
Comment (EdSchouten) [inline cmd/bb_browser/browser_service.go]:
```go
if authorization := req.Header.Get("Authorization"); authorization != "" {
    md.Set("authorization", authorization)
}
```

File: cmd/bb_browser/browser_service.go [hunk 2]
```diff
@@ -49,6 +50,15 @@ func getDigestFromRequest(req *http.Request) (*util.Digest, error) {
 		})
 }
 
+func extractContextFromRequest(req *http.Request) context.Context {
```
Comment (EdSchouten) [inline cmd/bb_browser/browser_service.go:55]: Could you add a comment above this function to explain what it does/what the purpose is?

**Review by EdSchouten — CHANGES_REQUESTED** [review]
Awesome!

**Comment by t-chaik** [comment]
Updated following your review advices.

**Pull request merged**